homeassistant/components/hive/light.py

```diff
@@ -116,35 +115,32 @@ def turn_on(self, **kwargs):
             get_new_color = kwargs.get(ATTR_HS_COLOR)
             hue = int(get_new_color[0])
             saturation = int(get_new_color[1])
-            new_color = (hue, saturation, self.brightness)
-
-        self.session.light.turn_on(
-            self.node_id,
-            self.light_device_type,
-            new_brightness,
-            new_color_temp,
-            new_color,
+            new_color = (hue, saturation, 100)
+
+        await self.hive.light.turn_on(
+            self.device, new_brightness, new_color_temp, new_color
         )
 
     @refresh_system
-    def turn_off(self, **kwargs):
+    async def async_turn_off(self, **kwargs):
         """Instruct the light to turn off."""
-        self.session.light.turn_off(self.node_id)
+        await self.hive.light.turn_off(self.device)
 
     @property
     def supported_features(self):
         """Flag supported features."""
         supported_features = None
-        if self.light_device_type == "warmwhitelight":
+        if self.device["hiveType"] == "warmwhitelight":
             supported_features = SUPPORT_BRIGHTNESS
-        elif self.light_device_type == "tuneablelight":
+        elif self.device["hiveType"] == "tuneablelight":
             supported_features = SUPPORT_BRIGHTNESS | SUPPORT_COLOR_TEMP
-        elif self.light_device_type == "colourtuneablelight":
+        elif self.device["hiveType"] == "colourtuneablelight":
             supported_features = SUPPORT_BRIGHTNESS | SUPPORT_COLOR_TEMP | SUPPORT_COLOR
 
         return supported_features
 
-    def update(self):
+    async def async_update(self):
         """Update all Node data from Hive."""
-        self.session.core.update_data(self.node_id)
-        self.attributes = self.session.attributes.state_attributes(self.node_id)
+        await self.hive.session.updateData(self.device)
+        self.device = await self.hive.light.get_light(self.device)
+        self.attributes.update(self.device.get("attributes", {}))
```

Comment:
If the attributes are always missing there will be a key error when looking up the attributes in `device_state_attributes`.

Reply:
Fixed in latest commit